address-space-controller/src/main/java/io/enmasse/controller/PodDistruptionBudgetController.java

```diff
@@ -0,0 +1,137 @@
+                    podDisruptionBudget.getSpec().setMaxUnavailable(new IntOrString(maxUnavailable));
+                    changed = true;
+                }
+
+                if (!routerSet.getStatefulSet().getSpec().getSelector().equals(podDisruptionBudget.getSpec().getSelector())) {
+                    podDisruptionBudget.getSpec().setSelector(routerSet.getStatefulSet().getSpec().getSelector());
+                    changed = true;
+                }
+
+                if (changed) {
+                    client.inNamespace(namespace).policy().podDisruptionBudget().createOrReplace(podDisruptionBudget);
+                }
+            } catch (KubernetesClientException e) {
+                log.warn("Error creating pod distruption budget for router", e);
+            }
+        }
+    }
+
+    private void reconcileBrokerPodDisruptionBudget(AddressSpace addressSpace, StandardInfraConfig infraConfig) {
+        if (infraConfig.getSpec() != null && infraConfig.getSpec().getBroker() != null &&
+                (infraConfig.getSpec().getBroker().getMinAvailable() != null || infraConfig.getSpec().getBroker().getMaxUnavailable() != null)) {
+
+            String name = String.format("enmasse.%s.%s.broker", addressSpace.getMetadata().getNamespace(), addressSpace.getMetadata().getName());
+            String infraUuid = addressSpace.getAnnotation(AnnotationKeys.INFRA_UUID);
+            try {
+                boolean changed = false;
+                PodDisruptionBudget podDisruptionBudget = client.inNamespace(namespace).policy().podDisruptionBudget().withName(name).get();
+                if (podDisruptionBudget == null) {
+                    podDisruptionBudget = new PodDisruptionBudgetBuilder()
+                            .editOrNewMetadata()
+                            .withName(name)
+                            .addToLabels("app", "enmasse")
+                            .endMetadata()
+                            .editOrNewSpec()
+                            .withNewSelector()
+                            .addToMatchLabels("app", "enmasse")
+                            .addToMatchLabels("role", "broker")
+                            .addToMatchLabels("infraType", "standard")
+                            .addToMatchLabels("infraUuid", infraUuid)
+                            .endSelector()
+                            .endSpec()
+                            .build();
+                    changed = true;
+                }
+
+                Integer minAvailable = infraConfig.getSpec().getRouter().getMinAvailable();
+                if (needsUpdate(podDisruptionBudget.getSpec().getMinAvailable(), minAvailable)) {
+                    podDisruptionBudget.getSpec().setMinAvailable(new IntOrString(minAvailable));
+                    changed = true;
+                }
+
+                Integer maxUnavailable = infraConfig.getSpec().getRouter().getMaxUnavailable();
```

Comment:
@lulf I believe this should be `.getBroker()` instead of `.getRouter()`

Reply:
You're right, thanks again.